sgt-back-end/server.ts

```diff
@@ -0,0 +1,151 @@
+import express from 'express';
+import { ClientError } from './client-error.js';
+import { errorMiddleware } from './error-middleware.js';
+import pg from 'pg';
+
+// only create ONE pool for your whole server
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.json());
+
+app.post('/api/grades', async (req, res, next) => {
+  const { name, course, score } = req.body;
+
+  if (
+    !name ||
+    !course ||
+    typeof score !== 'number' ||
+    score < 0 ||
+    score > 100
+  ) {
+    return next(
+      new ClientError(
```

Comment:
If you put this inside the try/catch, you'll be able to just throw instead of returning next().

People argue over whether try/catch should be broad around an entire function or small around just the code that needs it. With express, I find it best to have a try/catch that wraps the entire route handler. For what it's worth, I also prefer broader try/catch, because I find the try/catch to interrupt the readability of the code, so I like as few of them as possible.